addons/web/static/src/model/relational_model/static_list.js

```diff
@@ -814,9 +814,13 @@ export class StaticList extends DataPoint {
                 }
             } else if (command[0] === CREATE || command[0] === UPDATE) {
                 const record = this._cache[command[1]];
-                const values = record._getChanges(record._changes, { withReadonly });
-                if (command[0] === CREATE || Object.keys(values).length) {
-                    commands.push([command[0], command[1], values]);
+                if (command[0] === CREATE && record.resId) {
+                    commands.push([UPDATE, record.resId, record._getChanges(record._changes)]);
```

Comment:
```suggestion
                    // we created a new record, but it has already been saved (e.g. because we clicked
                    // on a view button in the x2many dialog), so replace the CREATE command by a
                    // LINK_TO
                    commands.push([UPDATE, record.resId, record._getChanges(record._changes)]);
```